kuma/javascript/src/document-provider.jsx

```diff
@@ -64,10 +48,11 @@ export default function DocumentProvider(
             throw new Error('DocumentProvider effect ran without body.');
         }
         const body = document.body;
+
         // This is the function that does client side navigation
-        function navigate(url, localeAndSlug) {
+        function navigate(url, slug) {
```

Comment:
A related change in this PR is that we never do client-side navigation if the locale is changing. So I can treat locale as a constant here, and just track the slug in the history mechanism. That's what a lot of the changes in this file are about.